assets/js/components/landing_page/overrides/UpgradeLegacyRecordButton.js

```diff
@@ -0,0 +1,97 @@
+// This file is part of InvenioRDM
+// Copyright (C) 2023 CERN.
+//
+// Zenodo RDM is free software; you can redistribute it and/or modify it
+// under the terms of the MIT License; see LICENSE file for more details.
+
+import React, { Component } from "react";
+import { Button, Grid, Message, Icon } from "semantic-ui-react";
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import { NewVersionButton } from "@js/invenio_rdm_records/";
+import PropTypes from "prop-types";
+import { http, withCancel } from "react-invenio-forms";
+
+export class UpgradeLegacyRecordButton extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      legacyRequestId: "",
+      fetchRequestsError: undefined,
+    };
+  }
+
+  componentDidMount() {
+    const { record, currentUserId } = this.props;
+    if (record.parent.access.owned_by[0].user == currentUserId) {
+      this.fetchRequests();
+    }
+  }
+
+  fetchRequests = async () => {
+    const { record } = this.props;
+    try {
+      const cancellableFetch = withCancel(
+        http.get(
+          `${record.links.requests}?q=type:legacy-record-upgrade&is_open=true`,
+          {
+            headers: {
+              Accept: "application/json",
+            },
+          }
+        )
+      );
+
+      const response = await cancellableFetch.promise;
+      const result = response.data.hits;
+      if (result.total > 0) {
+        this.setState({ legacyRequestId: result.hits[0].id });
+      }
+    } catch (error) {
+      let errorMessage = error.message;
+      if (error.response) {
+        errorMessage = error.response.data.message;
+      }
```

Comment:
why not using both error.message and error.response.data.message? why the latter is more important